python/ray/rllib/optimizers/sample_batch.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from functools import reduce
+
+import numpy as np
+
+
+class SampleBatch(object):
```

Comment:
TODO: needs to be consolidated with partial rollout somehow

Reply:
So this is a nice abstraction - perhaps it makes sense to extend PartialRollout from this which provides the ability to support truncated rollouts. If so, we can do it in a separate PR.